### ISSUE TITLE
Question On Impact Report Redirect

### ISSUE THREAD

**Issue by alexfinnarn** [CUCentralAdvancement/essential-cu#141]
Hi @Nicole Waldrip - fantastic question! I wouldn't use https://essential.cu.edu/impact-reports -- I think that redirects to this year's report because we haven't yet built a page that features all of them. We will, though, and that will likely be the URL. So, I'm looping in our lead developer @Alexander Finnarn here to see what solution would be best here.

Alex -- Nicole is asking if we have or could set up a URL that will always redirect to the current System impact report. She'd like to put it on a postcard they're creating, but the post card may be in use for multiple years so want to ensure the link always goes to most recent impact report and not just IR20.

I'm sure we could set up a unique, permanent URL to achieve this in relatively short order. Something like https://essential.cu.edu/impact-reports/latest is a common pattern, but we'll see what Alex thinks here.

Thank you!

Matt

**Comment by alexfinnarn** [comment]
@Matt Roush I like using a prefix or URL shortening service for these tasks and then updating that to point to the new report each time that is updated...or redesigning the homepage content could make sense. Some thoughts in order of preference would be:

- https://essential.cu.edu - Add a hero above-the-fold with a link to the current report.
- https://essential.cu.edu/r/impact-report - This keeps the "/r" pattern for redirects on that app.
- https://essential.cu.edu/ir - The "ir" path is unlikely to be used by any system routing and is very short only using two characters.
- https://essential.cu.edu/impact-report - This just drops the "s" and avoids adding "/latest".

My favorite thought is to have https://essential.cu.edu list a hero about the most recent report above-the-fold with a "Learn More" button leading to the full report. Currently, that page's above-the-fold content is lacking IMHO and there is space to fit a blurb like this. Since the site mainly hosts impact reports, it makes sense to feature them on the homepage...in fact, there is not a great way to navigate to past reports ATM.

Whatever the case, I think the redirect should be updated manually rather than trying a fancy route resolver. The maintenance is basically changing a line in a JSON file for redirects once a year. Going with the homepage content update, then the maintenance would be coming up with a new hero (image, description, button text) each year but can still be data in a JSON file for easier maintenance or worked into the CMS.

-Alex

**Comment by alexfinnarn** [comment]
Thanks, Alex. So given that they probably need this sooner than later, how about we do https://essential.cu.edu/impact-report (no "s", no "/r/") and we'll just update the route manually each year.

I think something on the essential.cu.edu homepage would be good in the future, but not something we'll be able to do quickly as many people have a say in that (incl. Tim, who is out all week).

Thanks,
Matt

**Comment by alexfinnarn** [comment]
Alright, I think that makes sense then. This requires a code change to complete and can be moved to GitHub now in https://github.com/CUCentralAdvancement/essential-cu/issues

**Comment by alexfinnarn** [comment]
https://github.com/CUCentralAdvancement/essential-cu/commit/ec0f93251b06793fb7795cddece5eca502df3819

That is the commit that added the redirect, but the issue, #83, doesn't have info that I can see on that specific redirect.

**Comment by alexfinnarn** [comment]
@mattroush Since https://essential.cu.edu/impact is used to complete the Workfront ticket, is this issue still valid? Any followup?

**Comment by mattroush** [comment]
Let's just make sure it's set up for us to easily update each year and then document it's existence, purpose and how to update it somewhere. Other than that, I think we're good.

**Comment by alexfinnarn** [comment]
I added a section to the readme https://github.com/CUCentralAdvancement/essential-cu#maintenance I wrote it in the example of standardized info ever project readme would have...definitions of the status, audience, dependencies, etc. will be put in another doc and then each time a repo is made, the readme includes stubs as a template.

I'm not sure the history of all the redirects, but discussions about those, if necessary, should go into another issue.
